src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -277,14 +301,58 @@ object DataTypes {
       predicate.map(_.into[Predicate].withFieldConst(_.field, field).transform)
   }
 
+  /** Class representing snapshot */
+  case class Snapshot(field: String, value: Any) {
+
+    /** Transforms snapshot to predicate */
+    def toPredicate: Predicate =
+      Predicate(
+        field = field,
+        operation = OperationType.after,
+        set = List(value),
+        inverse = true
+      )
+
+    /** Transforms snapshot to ordering */
+    def toOrdering: QueryOrdering =
+      QueryOrdering(
+        field = field,
+        direction = OrderDirection.desc
+      )
+  }
+
+  /** Finds invalid snapshot fields*/
+  def findInvalidSnapshotFields(
+      query: Query,
+      entity: EntityPath,
+      metadataConfiguration: MetadataConfiguration
+  ): List[InvalidSnapshotField] =
+    query.snapshot match {
+      case Some(snapshot) =>
+        metadataConfiguration.entity(entity).flatMap { entityCfg =>
+          entityCfg.attributes.filter {
```

Comment:
Maybe I'm wrong here: is it filtering by value type and then looking for the right key?

Wouldn't it be simpler then to do a single `.find` that verifies both predicates, on key and value?

Reply:
sure, changed